### PR TITLE
Added Remove Debug Output Option

### PR DESCRIPTION
**Pull request by TheFocusMan**
If the option enabled the compiler run faster,
The fast boost also remove more than 1000 lines of code in the final asm file

### CODE REVIEW

**Comment by TheFocusMan** [comment]
> This should have a different name other than fast boost, it's not proper English

The English matter?

**Comment by TheFocusMan** [comment]
> > > This should have a different name other than fast boost, it's not proper English
> > 
> > 
> > The English matter? if the function is work
> 
> Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt

I dont have a better way to name this

**Comment by terminal-cs** [comment]
> > > > This should have a different name other than fast boost, it's not proper English
> > > 
> > > 
> > > The English matter? if the function is work
> > 
> > 
> > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> 
> I dont have a better way to name this

name it EnableOptimizations or something

**Comment by TheFocusMan** [comment]
> > > > > This should have a different name other than fast boost, it's not proper English
> > > > 
> > > > 
> > > > The English matter? if the function is work
> > > 
> > > 
> > > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> > 
> > 
> > I dont have a better way to name this
> 
> name it EnableOptimizations or something
I renamed it

**Comment by terminal-cs** [comment]
> > > > > > This should have a different name other than fast boost, it's not proper English
> > > > > 
> > > > > 
> > > > > The English matter? if the function is work
> > > > 
> > > > 
> > > > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> > > 
> > > 
> > > I dont have a better way to name this
> > 
> > 
> > name it EnableOptimizations or something
> 
> I renamed it

i meant in the code

**Comment by TheFocusMan** [comment]
> > > > > > > This should have a different name other than fast boost, it's not proper English
> > > > > > 
> > > > > > 
> > > > > > The English matter? if the function is work
> > > > > 
> > > > > 
> > > > > Yes, it matters how things are spelled. the code will become messy if you don't care how it's spelt
> > > > 
> > > > 
> > > > I dont have a better way to name this
> > > 
> > > 
> > > name it EnableOptimizations or something
> > 
> > 
> > I renamed it
> 
> i meant in the code

i renamed in the code

**Comment by terminal-cs** [comment]
why littleoptimization? that doesn't make too much sense in english either
why not just Optimization?

File: source/Cosmos.IL2CPU/AppAssembler.cs [hunk 1]
```diff
@@ -1371,7 +1371,7 @@ public void EmitEntrypoint(MethodBase aEntrypoint, MethodBase[] aBootEntries = n
             }
             else
             {
-                X86.IL.Call.DoExecute(Assembler, null, aEntrypoint.DeclaringType.BaseType.GetMethod(CompilerEngine.UseGen3Kernel ? "EntryPoint" : "Start"), null, xCurLabel, CosmosAssembler.EntryPointName + ".AfterStart", DebugEnabled);
```
Comment (quajak) [inline source/Cosmos.IL2CPU/AppAssembler.cs:1374]: Why did you change this?

**Review by quajak — CHANGES_REQUESTED** [review]
Thanks for the contribution!

Regarding naming, can you please name this `BootDebugOutput` or similar?

File: source/Cosmos.IL2CPU/AppAssembler.cs (hunk 1 above)
Comment (TheFocusMan) [reply source/Cosmos.IL2CPU/AppAssembler.cs:1374]: To override Kernel.Start

**Comment by TheFocusMan** [comment]
> Thanks for the contribution!
> 
> Regarding naming, can you please name this `BootDebugOutput` or similar?

I renamed it,
To get the code work all the three pull requests need to be merged

File: source/Cosmos.IL2CPU/AppAssembler.cs (hunk 1 above)
Comment (quajak) [reply source/Cosmos.IL2CPU/AppAssembler.cs:1374]: I see, that is a sensible change. Can you add a sentence to https://github.com/CosmosOS/Cosmos/blob/master/Docs/articles/Kernel/Startup.md/ so that people are aware of the possibility?

**Review by quajak — APPROVED** [review]
The change looks good. Please add documentation for the change to the entry method determination. We can merge this when the other two are ready.

**Comment by TheFocusMan** [comment]
> The change looks good. Please add documentation for the change to the entry method determination. We can merge this when the other two are ready.

Ok now i added a sentence that explain to people can override Kernel.Start()

**Comment by quajak** [comment]
Thanks
